### DISCUSSION TITLE
The switch -c alone don't seems to have any effect + convert to 8.1 compatible is mandatory ?

### DISCUSSION THREAD

**Discussion by Emilius256** [quietvoid/dovi_tool#68] [Q&A] [answered]
Hello,

I did my first encode using your tool, and it seems that using the option `-c` or `--crop` alone does not have any effect, the generated RPU file with or without this option have the same size and the same MD5 hash, and the encoded file is "overcropped" with both files.
While ti seems to work correctly if used together with the `-m` option.
It is a bug or the `-c` option must be used together with `-m` ?
The command line used:
`dovi_tool.exe -c extract-rpu source_EL.265 -o output_RPU.bin` no effect
`dovi_tool.exe -m 0 -c extract-rpu source_EL.265 -o output_RPU.bin` or `dovi_tool.exe -m 2 -c extract-rpu source_EL.265 -o output_RPU.bin` works as expected

One last question, if I encode using x265 with the parameters `--dolby-vision-profile 8.1 --dolby-vision-rpu RPU.bin` it is mandatory to use the option `-m 2` ?

Thank you for your help and for your tool.

**Comment by quietvoid** [comment, marked as answer]
Using `-c` doesn't really make sense unless the RPU is modified as well.
But I guess it should imply at least `-m 0`, so that profile 8 files can be cropped without further modification.


> One last question, if I encode using x265 with the parameters --dolby-vision-profile 8.1 --dolby-vision-rpu RPU.bin it is mandatory to use the option -m 2 ?

I wouldn't say mandatory but mode 2 is meant to output a profile 8.1 RPU, which is what x265 expects.

**Reply by Emilius256** [reply]
Thank you very much for your answer.

**Reply by quietvoid** [reply]
I've implemented the default fallback to mode 0 when cropping: 81d30e0919006e815901d43f3e945ff1505a5884
